network/stream/sync_provider.go

```diff
@@ -50,6 +56,8 @@ type syncProvider struct {
 	quit                    chan struct{}     // shutdown
 	cacheMtx                sync.RWMutex      // synchronization primitive to protect cache
 	cache                   *lru.Cache        // cache to minimize load on netstore
+	setCacheMtx             sync.RWMutex      // set cache mutex
+	setCache                *lru.Cache        // cache to reduce load on localstore to not set the same chunk as synced
```

Comment:
I think that a cache with ttl would fit much better than lru cache. Eventually, the whole cache will be filled up and stay at that level, and the cache size of 800MB is not insignificant.

Reply:
There is a chance, less with larger cache size, but if that happens, the consequence would be that the chunk will be potentially set twice or multiple times, which does not influence the state, just performance.